src/icrar/leap-accelerate/core/resource_usage.cc

```diff
@@ -0,0 +1,86 @@
+/**
+ * ICRAR - International Centre for Radio Astronomy Research
+ * (c) UWA - The University of Western Australia
+ * Copyright by UWA(in the framework of the ICRAR)
+ * All rights reserved
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111 - 1307  USA
+ */
+
+#include <chrono>
+#include <iostream>
+#include <sys/time.h>
+#include <sys/resource.h>
+
+#include "icrar/leap-accelerate/core/ioutils.h"
+#include "icrar/leap-accelerate/core/logging.h"
+#include "icrar/leap-accelerate/core/profiling_timer.h"
+#include "icrar/leap-accelerate/core/resource_usage.h"
+#include "icrar/leap-accelerate/exception/exception.h"
+
+namespace icrar {
+
+static profiling_timer walltime_timer;
+
+static usec_t to_usecs(const struct timeval &t)
+{
+    return t.tv_sec * 1000000 + t.tv_usec;
+}
+
+/// Returns the maximum Resident Storage Size of this process
+/// (i.e., the maximum amountof memory used).
+ResourceUsage get_resource_usage()
```

Comment:
Move this to a private method within UsageReporter

Reply:
This is related to my comment above -- I see this as useful on its own, so I'd vote for leaving it publicly exposed for easy reuse.